src/write.c

```diff
@@ -402,8 +396,23 @@ static uint8_t avifItemPropertyDedupFinish(avifItemPropertyDedup * dedup, avifRW
     property->index = ++dedup->nextIndex; // preincrement so the first new index is 1 (as ipma is 1-indexed)
     property->size = newPropertySize;
     property->offset = avifRWStreamOffset(outputStream);
-    avifRWStreamWrite(outputStream, dedup->buffer.data, newPropertySize);
-    return property->index;
+    *propertyIndex = property->index;
+    AVIF_CHECKRES(avifRWStreamWrite(outputStream, dedup->buffer.data, newPropertySize));
+    return AVIF_RESULT_OK;
+}
+
+static avifResult avifItemPropertyDedupFinishAndIpmaPush(avifItemPropertyDedup * dedup,
+                                                         avifRWStream * outputStream,
+                                                         struct ipmaArray * ipma,
+                                                         avifBool essential)
+{
+    uint8_t propertyIndex;
+    AVIF_CHECKRES(avifItemPropertyDedupFinish(dedup, outputStream, &propertyIndex));
+    AVIF_CHECKERR(ipma->count < MAX_ASSOCIATIONS, AVIF_RESULT_UNKNOWN_ERROR);
```

Comment:
[Just a comment, no change requested.] When I reviewed the original code, I verified this property by counting the number of `ipmaPush()` calls. It's certainly better to let the code check this property.